tests/cnf/ran/ptp/tests/ptp-events-and-metrics.go

```diff
@@ -0,0 +1,165 @@
+package tests
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/glog"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	prometheusv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	eventptp "github.com/redhat-cne/sdk-go/pkg/event/ptp"
+	"github.com/rh-ecosystem-edge/eco-goinfra/pkg/reportxml"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/internal/querier"
+	. "github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/consumer"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/events"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/iface"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/metrics"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/profiles"
+	"github.com/rh-ecosystem-edge/eco-gotests/tests/cnf/ran/ptp/internal/tsparams"
+)
+
+var _ = Describe("PTP Events and Metrics", Label(tsparams.LabelEventsAndMetrics), func() {
+	var prometheusAPI prometheusv1.API
+
+	BeforeEach(func() {
+		By("creating a Prometheus API client")
+		var err error
+		prometheusAPI, err = querier.CreatePrometheusAPIForCluster(RANConfig.Spoke1APIClient)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create Prometheus API client")
+
+		By("ensuring clocks are locked before testing")
+		err = metrics.AssertQuery(context.TODO(), prometheusAPI, metrics.ClockStateQuery{}, metrics.ClockStateLocked,
+			metrics.AssertWithStableDuration(10*time.Second),
+			metrics.AssertWithTimeout(5*time.Minute))
+		Expect(err).ToNot(HaveOccurred(), "Failed to assert clock state is locked")
+	})
+
```

Comment:
Shall we also add AfterEach to revert config and check clock state in general? Or perhaps add to the last two test cases only.